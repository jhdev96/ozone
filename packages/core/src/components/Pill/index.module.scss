@import '../../scss/root';
@import '../../scss/mixins/pill/pill.scss';


/* Main pill classes */
.success {
  @include pill;

  color: var(--green-3);
  background-color: var(--green-1);
}

.info {
  @include pill;

  color: var(--blue-6);
  background-color: var(--blue-2);
}

.warning {
  @include pill;

  color: var(--yellow-4);
  background-color: var(--yellow-1);
}

.error {
  @include pill;

  color: var(--red-4);
  background-color: var(--red-1);
}

.routine {
  @include pill;

  color: var(--black-2);
  background-color: var(--gray-core);
}