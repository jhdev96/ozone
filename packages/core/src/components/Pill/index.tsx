import React from 'react';
import styles from './index.module.scss';


interface PillProps {
  /** 
   * Main classes for pill components.  
   */
  variation?: "success" | 
              "info"    | 
              "warning" | 
              "error"   | 
              "routine";
  /** 
   * The content rendered inside a pill. 
   */
  children?: React.ReactNode
}

export default function Pill({
  variation = "routine", 
  children
}: PillProps): JSX.Element 
{
  return (
    <span className={styles[variation]}>
      {children}
    </span>
  )
}

Pill.displayName = 'Pill';