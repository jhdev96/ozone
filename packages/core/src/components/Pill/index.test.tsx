import React from "react";
import { render } from "@testing-library/react";
import Pill from '.';


describe('<Pill />', () => {
  it('renders a pill with text', () => {
    const {container} = render(<Pill>text</Pill>);

    expect(container).toHaveTextContent("text");
  });

  it('renders a pill of type `routine` by default', () => {
    const {container} = render(<Pill>text</Pill>);
    
    expect(container).toMatchSnapshot();
  });

  it('renders the correct variations', () => {
    const {container} = render(
      <>
        <Pill variation="info">text</Pill>
        <Pill variation="success">text</Pill>
        <Pill variation="warning">text</Pill>
        <Pill variation="error">text</Pill>
        <Pill variation="routine">text</Pill>
      </>
    );

    expect(container).toMatchSnapshot();
  });
});