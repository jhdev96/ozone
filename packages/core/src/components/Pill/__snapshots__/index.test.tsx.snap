// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill /> renders a pill of type \`routine\` by default 1`] = `
<div>
  <span
    class="routine"
  >
    text
  </span>
</div>
`;

exports[`<Pill /> renders the correct variations 1`] = `
<div>
  <span
    class="info"
  >
    text
  </span>
  <span
    class="success"
  >
    text
  </span>
  <span
    class="warning"
  >
    text
  </span>
  <span
    class="error"
  >
    text
  </span>
  <span
    class="routine"
  >
    text
  </span>
</div>
`;
